### ISSUE TITLE
Cloud Provider information is no longer available

### ISSUE THREAD

**Issue by cooktheryan** [font/pacman#6]
We recently patched our OCP cluster and discovered the ability to query the cloud provider metadata is not longer allowed. It appears this access is blocked. This was confirmed with the network team that containers no longer have access to this information.

We are doing a work around of this by using the kubeapi. We create a service account for pacman, cluster role, cluster role binding and pass in the environment variable of the node name to be used when querying information through the kubeapi.

SA
```
apiVersion: v1
kind: ServiceAccount
metadata:
  name: pacman 
```
ClusterRole and ClusterRoleBinding
```
apiVersion: rbac.authorization.k8s.io/v1beta1
kind: ClusterRole
metadata:
  name: pacman
rules:
- apiGroups: [""]
  resources: ["pods", "nodes"]
  verbs: ["get", "watch", "list"]

---
apiVersion: rbac.authorization.k8s.io/v1beta1
kind: ClusterRoleBinding
metadata:
  name: pacman
  namespace: pacmanv3
subjects:
- kind: ServiceAccount
  name: pacman 
  namespace: pacmanv3
roleRef:
  kind: ClusterRole
  name: pacman
  apiGroup: rbac.authorization.k8s.io
```

Deployment
```
apiVersion: extensions/v1beta1
kind: Deployment
metadata:
  namespace: pacmanv3
  name: pacman
  labels:
    name: pacman
spec:
  replicas: 1
  selector:
    matchLabels:  
      name: pacman
  template:
    metadata:
      namespace: pacman
      labels:
        name: pacman
    spec:
      serviceAccount: pacman
      containers:
      - name: pacman
        image: quay.io/rcook/pacman-nodejs-app:latest
        imagePullPolicy: Always
        env:
        - name: MONGO_SERVICE_HOST
        # Single member MongoDB
          value: internal-a4dfda3218e1e11e9967c0edf0f025f2-791543394.us-east-1.elb.amazonaws.com
        # Replicaset (federated) MongoDB
        #  value: mongo-east1.apps.east-1.sysdeseng.com,mongo-east2.apps.east-2.sysdeseng.com,mongo-west2.apps.west-2.sysdeseng.com
        # Comment out MONGO_REPLICA_SET for the single-member Mongo
        # - name: MONGO_REPLICA_SET
        #  value: rs0
        - name: MONGO_AUTH_USER
          valueFrom:
            secretKeyRef:
              key: database-user
              name: mongodb-users-secret
        - name: MONGO_AUTH_PWD
          valueFrom:
            secretKeyRef:
              key: database-password
              name: mongodb-users-secret
        - name: MONGO_DATABASE
          value: pacman
        - name: MONGO_PORT
          value: "27017"
        - name: MONGO_USE_SSL
          value: "false"
        - name: MONGO_VALIDATE_SSL
          value: "false"
        - name: MY_NODE_NAME
          valueFrom:
            fieldRef:
              fieldPath: spec.nodeName
        ports:
        - containerPort: 8080 
```

This is the location.js we are working with now to try to redo the cloud provider check https://github.com/dav1x/pacman/blob/master/routes/location.js

This is the rough curl command which will allow for us to get the data that we need.
```
curl -v --cacert /var/run/secrets/kubernetes.io/serviceaccount/ca.crt -H "Authorization: Bearer $(cat /var/run/secrets/kubernetes.io/serviceaccount/token)" https://kubernetes.default.svc/api/v1/nodes/$MY_NODE_NAME 
```

**Comment by font** [comment]
Thanks @cooktheryan! Do you know if this is specific to OpenShift? If not, we may need to make #7 the only default method of querying for cloud metadata.

**Comment by cooktheryan** [comment]
@font I have not had the ability to check. I will try to get a kube cluster up to validate

**Comment by font** [comment]
@cooktheryan @dav1x I've merged #7, built and pushed a new image `quay.io/ifont/pacman-nodejs-app:latest`. Thanks for identifying and fixing the issue! Feel free to close this issue if you feel it's been adequately addressed.
